rkt/status.go

```diff
@@ -16,22 +16,156 @@
 
 package main
 
-import "fmt"
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/appc/spec/schema/types"
+	"github.com/coreos/rocket/pkg/lock"
+)
 
 var (
 	cmdStatus = &Command{
 		Name:    "status",
 		Summary: "Check the status of a rkt job",
-		Usage:   "UUID",
+		Usage:   "--uuid UUID [--wait]",
 		Run:     runStatus,
 	}
+	flagContainerUUID types.UUID
+	flagWait          bool
+)
+
+const (
+	statusDir = "stage1/rkt/status"
 )
 
 func init() {
 	commands = append(commands, cmdStatus)
+	cmdStatus.Flags.Var(&flagContainerUUID, "uuid", "uuid of container to query the status of")
+	cmdStatus.Flags.BoolVar(&flagWait, "wait", false, "toggle waiting for the container to exit if running")
 }
 
 func runStatus(args []string) (exit int) {
-	fmt.Println("Not implemented.")
+	if flagContainerUUID.Empty() {
+		fmt.Fprintf(os.Stderr, "--uuid required\n")
+		return 1
+	}
+
+	isGarbage := false
+	cid := flagContainerUUID.String()
+
+	cp := filepath.Join(containersDir(), cid)
+	l, err := lock.NewLock(cp)
+	if err != nil {
+		cp = filepath.Join(garbageDir(), cid)
+		l, err = lock.NewLock(cp)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unable to open lock for container %q: %v\n", cid, err)
```

Comment:
run barely uses the log package, but I'll put together a separate commit later converting all the fmt.Fprintf's to log.Printfs throughout the code to get it consistent.